app/domain/invoice/qrcode.rb

```diff
@@ -119,37 +144,72 @@ def generate_png # rubocop:disable Metrics/MethodLength
     )
   end
 
-  def extract_contact(contactable) # rubocop:disable Metrics/MethodLength
-    parts = contactable.to_s.strip.split(/\r*\n/)
-    address_line1 = nil
-    address_line2 = nil
-    if parts.count > 1
-      address_line1 = parts.last
-    end
-    if parts.count > 2
-      address_line2 = address_line1
-      address_line1 = parts.second_to_last
-    end
+  def striped_values(*hashes)
+    hashes.collect { |obj| obj.values.collect(&:to_s).collect(&:strip) }
+  end
+
+  def image(filename)
+    Rails.root.join("app/domain/invoice/assets/#{filename}")
+  end
+
+  def show_total?
+    !@invoice.hide_total? && @invoice.total.nonzero?
+  end
+
+  def format_address(address_attrs)
+    if address_attrs[:address_type] == "K"
+      [
+        address_attrs[:name],
+        address_attrs[:address_line1],
+        address_attrs[:address_line2]
+      ]
+    else
+      [
+        address_attrs[:name],
+        [address_attrs[:street], address_attrs[:housenumber]].compact.join(" "),
+        [address_attrs[:zip_code], address_attrs[:town]].compact.join(" ")
+      ]
+    end.compact_blank.join("\n")
+  end
+
+  def deprecated_creditor
```

Comment:
Ich fänd auch noch gut, wenn alles was mit deprecation zu tun hat in einem eigenen Objekt versorgt wäre, zb. `Invoice::QrCode::Deprecations`